yarn-project/world-state/src/world-state-db/merkle_trees.ts

```diff
@@ -64,7 +92,8 @@ class PublicDataTree extends StandardIndexedTree {
  * A convenience class for managing multiple merkle trees.
  */
 export class MerkleTrees implements MerkleTreeDb {
-  private trees: (AppendOnlyTree | UpdateOnlyTree)[] = [];
+  // gets initialized in #init
+  private trees: MerkleTreeMap = null as any;
```

Comment:
I'd move the initialization logic to the static `new` method, and have the constructor receive the `MerkleTreeMap` already initialized, so we don't have to do this hack.